src/librustdoc/html/render.rs

```diff
@@ -1348,7 +1346,7 @@ impl<'a> SourceCollector<'a> {
             cur.push(component);
             fs::create_dir_all(&cur).unwrap();
             root_path.push_str("../");
-            href.push_str(component);
+            href.push_str(&component.to_string_lossy());
```

Comment:
Why not `.to_str().unwrap()` (as before in `clean_srcpath`)? It will only be a dead link if replacement characters are inserted.

Reply:
because I saw the same thing being done [a few lines below](https://github.com/rust-lang/rust/blob/ddab10a692aab2e2984b5c826ed9d78a57e94851/src/librustdoc/html/render.rs#L1367) so I assumed that was expected outcome